src/irmin-pack/unix/control_file_intf.ml

```diff
@@ -103,7 +103,19 @@ module Payload_v3 = struct
       extensions *)
 end
 
-module Latest_payload = Payload_v3
+module Payload_v4 = struct
+  include Payload_v3
+
+  type t = {
+    dict_end_poff : int63;
+    suffix_end_poff : int63;
+    status : Payload_v3.status;
+    commit_offset_in_prefix : int63;
```

Comment:
You need to create a new `status` type. The commit offset is not available prior to the first GC. This will be better than setting that to `Int63.zero`.

Reply:
I'm not sure what you are suggesting. 
For status:
```
 type status =
    | From_v1_v2_post_upgrade of from_v1_v2_post_upgrade
    | From_v3_no_gc_yet
    | From_v3_used_non_minimal_indexing_strategy
    | From_v3_gced of from_v3_gced
    | From_v4_no_gc_yet
    | From_v4_used_non_minimal_indexing_strategy
    | From_v4_gced of older_live_commit
    | ....
```
that is you want to have the `older_live_commit` part of the status and keep the `t` unchanged?

Also for the chunk files then (since I'll do that next), it should be 
 ```
    ...
    | From_v4_gced of (older_live_commit * chunk_info)
```

Reply:
I forgot to check the planning notes for this.
 
I added a new status in V4, still not sure if this is what you had in mind. In the doc, you talk about a minor for version 3, but since we have version 4 in the issue, I imagine that the idea of minor version is abandoned?